pyrit/cli/frontend_core.py

```diff
@@ -0,0 +1,738 @@
+
+    Raises:
+        ValueError: If database type is invalid.
+    """
+    valid_databases = [IN_MEMORY, SQLITE, AZURE_SQL]
+    if database not in valid_databases:
+        raise ValueError(f"Invalid database type: {database}. " f"Must be one of: {', '.join(valid_databases)}")
+    return database
+
+
+def validate_log_level(*, log_level: str) -> str:
+    """
+    Validate log level.
+
+    Args:
+        log_level: Log level string (case-insensitive).
+
+    Returns:
+        Validated log level in uppercase.
+
+    Raises:
+        ValueError: If log level is invalid.
+    """
+    valid_levels = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
+    level_upper = log_level.upper()
+    if level_upper not in valid_levels:
+        raise ValueError(f"Invalid log level: {log_level}. " f"Must be one of: {', '.join(valid_levels)}")
+    return level_upper
+
+
+def validate_integer(value: str, *, name: str = "value", min_value: Optional[int] = None) -> int:
+    """
+    Validate and parse an integer value.
+
+    Note: The 'value' parameter is positional (not keyword-only) to allow use with
+    argparse lambdas like: lambda v: validate_integer(v, min_value=1).
+    This is an exception to the PyRIT style guide for argparse compatibility.
+
+    Args:
+        value: String value to parse.
+        name: Parameter name for error messages. Defaults to "value".
+        min_value: Optional minimum value constraint.
+
+    Returns:
+        Parsed integer.
+
+    Raises:
+        ValueError: If value is not a valid integer or violates constraints.
+    """
+    try:
+        int_value = int(value)
+    except (ValueError, TypeError) as e:
+        raise ValueError(f"{name} must be an integer, got: {value}") from e
+
+    if min_value is not None and int_value < min_value:
+        raise ValueError(f"{name} must be at least {min_value}, got: {int_value}")
+
+    return int_value
+
+
+def _argparse_validator(validator_func):
+    """
+    Decorator to convert ValueError to argparse.ArgumentTypeError.
+
+    This decorator adapts our keyword-only validators for use with argparse's type= parameter.
+    It handles two challenges:
+
+    1. Exception Translation: argparse expects ArgumentTypeError, but our validators raise
+       ValueError. This decorator catches ValueError and re-raises as ArgumentTypeError.
+
+    2. Keyword-Only Parameters: PyRIT validators use keyword-only parameters (e.g.,
+       validate_database(*, database: str)), but argparse's type= passes a positional argument.
+       This decorator inspects the function signature and calls the validator with the correct
+       keyword argument name.
+
+    This pattern allows us to:
+    - Keep validators as pure functions with proper type hints
+    - Follow PyRIT style guide (keyword-only parameters)
+    - Reuse the same validation logic in both argparse and non-argparse contexts
+
+    Args:
+        validator_func: Function that raises ValueError on invalid input.
```

Comment:
```suggestion
        validator_func (Callable): Function that raises ValueError on invalid input.
```
functools has some utilities that might help here, but I'm not too familiar